src/licensedcode/plugin_license.py

```diff
@@ -116,3 +119,57 @@ def get_scanner(
             license_text_diagnostics=license_text_diagnostics,
             license_url_template=license_url_template
         )
+
+    def process_codebase(self, codebase, **kwargs):
+        
+        if codebase.has_single_resource:
+            return
+
+        for resource in codebase.walk(topdown=False):
+            match_reference_license(resource,codebase)
+
+
+def match_reference_license(resource, codebase):
+    """
+    Find instances for any licenses in referenced filenames
+    """
+    if not resource.is_file:
+        return
+
+    licenses = resource.licenses
+    license_expressions = resource.license_expressions
+    if not licenses:
+        return 
+
+    reference_licenses=[]
+    reference_license_expressions=[]
+    unique_files = []
+
+    for license in licenses:
+        ref_files=license['matched_rule']['referenced_filenames']
+        if len(ref_files) != 0:
+            for i in range(len(ref_files)):
+                if ref_files[i].startswith('usr/share/common-licenses') or ref_files[i] in unique_files:
+                    continue
+                unique_files.append(ref_files[i])
+                new_resource = find_reference_licenses(ref_files[i], resource, codebase)
+                if new_resource:
+                    reference_licenses.extend(new_resource.licenses)
+                    reference_license_expressions.extend(new_resource.license_expressions)
+
+    licenses.extend(reference_licenses)
+    license_expressions.extend(reference_license_expressions)
+    codebase.save_resource(resource)
+    return resource
+
+
+def find_reference_licenses(referenced_filename, resource, codebase):
+    """
+    Returns the resource which references to referenced_filename if found
```

Comment:
Use imperative style `Return` not `Returns`
And `Returns the resource which references to referenced_filename if found` is hard to understand. I am not sure what you return... is this a `Resource` object? What are the three arguments you pass `referenced_filename`, `resource`, `codebase`?
One suggestion could be
```suggestion
    Return a Resource matching the `referenced_filename` path or None, given a ``resource`` Resource (what is this resource... you need to explain in a ``codebase`` Codebase.
```

Reply:
Also not sure why the resource object is being returned, it is already saved in the codebase.